index.html

```diff
@@ -919,13 +918,285 @@ <h4>Verify Proof (eddsa-jcs-2022)</h4>
+<em>transformed data document</em> and <em>proof configuration</em>
+into cryptographic hash data that is ready to be provided as input to the
+algorithms in Section <a href="#proof-serialization-eddsa-jcs-2022"></a> or
+Section <a href="#proof-verification-eddsa-jcs-2022"></a>.
+          </p>
+
+          <p>
+The required inputs to this algorithm are a <em>transformed data document</em>
+(<var>transformedDocument</var>) and <em>canonical proof configuration</em>
+(<var>canonicalProofConfig</var>). A single <em>hash data</em> value represented as
+series of bytes is produced as output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Let <var>transformedDocumentHash</var> be the result of applying the
+SHA-256 (SHA-2 with 256-bit output) cryptographic hashing algorithm [[RFC6234]]
+to the <var>transformedDocument</var>. <var>transformedDocumentHash</var> will
+be exactly 32 bytes in size.
+            </li>
+            <li>
+Let <var>proofConfigHash</var> be the result of applying the
+SHA-256 (SHA-2 with 256-bit output) cryptographic hashing algorithm [[RFC6234]]
+to the <var>canonicalProofConfig</var>. <var>proofConfigHash</var> will be
+exactly 32 bytes in size.
+            </li>
+            <li>
+Let <var>hashData</var> be the result of joining <var>proofConfigHash</var> (the
+first hash) with <var>transformedDocumentHash</var> (the second hash).
+            </li>
+            <li>
+Return <var>hashData</var> as the <em>hash data</em>.
+            </li>
+          </ol>
+
+        </section>
+        <section>
+          <h4>Proof Configuration (eddsa-jcs-2022)</h4>
+
+          <p>
+The following algorithm specifies how to generate a
+<em>proof configuration</em> from a set of <em>proof options</em>
+that is used as input to the <a href="#hashing-eddsa-jcs-2022">proof hashing algorithm</a>.
+          </p>
+
+          <p>
+The required inputs to this algorithm are the <em>document</em>
+(|unsecuredDocument|) and the <em>proof options</em>
+(<var>options</var>). The <em>proof options</em> MUST contain a type identifier
+for the
+<a data-cite="vc-data-integrity#dfn-cryptosuite">
+cryptographic suite</a> (<var>type</var>) and MUST contain a cryptosuite
+identifier (<var>cryptosuite</var>). A <em>proof configuration</em>
+object is produced as output.
+          </p>
+
+          <ol class="algorithm">
+            <li>
+Let <var>proofConfig</var> be an empty object.
+            </li>
+            <li>
+Set <var>proofConfig</var>.<var>type</var> to
+<var>options</var>.<var>type</var>.
+            </li>
+            <li>
+If <var>options</var>.<var>cryptosuite</var> is set, set
+<var>proofConfig</var>.<var>cryptosuite</var> to its value.
+            </li>
+            <li>
+If <var>options</var>.<var>type</var> is not set to `DataIntegrityProof` and
+<var>proofConfig</var>.<var>cryptosuite</var> is not set to `eddsa-jcs-2022`, an
+`INVALID_PROOF_CONFIGURATION` error MUST be raised.
```

Comment:
This needs to become `MALFORMED_PROOF_ERROR` and use similar language to the above text. We probably also just want to run the checks on `proofConfig` since it's now a clone. Similar to this:

https://w3c.github.io/vc-di-ecdsa/#proof-configuration-ecdsa-rdfc-2019

Though the text at that link still needs the error language updates / reconciliation with what was fixed with the interface in the data integrity core spec. If necessary, we can just do those pieces in a separate PR, but we do need to bring the error language inline.

Reply:
I'm a bit confused the link above has a `INVALID_PROOF_CONFIGURATION` error. Is that supposed to be a `MALFORMED_PROOF_ERROR`? Then there is language changes for error handling that needed to be check for all specifications right? Maybe get this PR done then scrub through everything, unless this is the only change needed.

Reply:
I forgot is the error language in the DI spec the latest, e.g. "an error MUST be raised and SHOULD convey an error type of [PROOF_GENERATION_ERROR](https://w3c.github.io/vc-data-integrity/#PROOF_GENERATION_ERROR)."

And we want to make sure all cryptosuite documents conform and use listed error codes?